scripts/controller.gd

```diff
@@ -64,12 +65,14 @@ func _on_ws_received_message(_msg_type: String) -> void:
 		match msg["type"]:
 			DSGMessageType.ROUND_START:
 				if self._current_state == STATES.AWAITING_ROUND:
-					self._start_round(msg["payload"])
 					self._current_state = STATES.ONGOING_ROUND
+					self._start_round(msg["payload"])
 			DSGMessageType.ROUND_RESULT:
 				if self._current_state == STATES.AWAITING_RESULT:
-					self._animate_round(msg["payload"])
 					self._current_state = STATES.ONGOING_ANIMATION
+					self._animate_round(msg["payload"])
+			DSGMessageType.ERROR:
```

Comment:
The `MSG_FILTER` currently makes this unreachable. I think we should catch and log the error in `DSGNetwork` to make sure they're always logged, regardless of the state.

Reply:
18a9ee2 - errors are currently not logged (todo)